modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4024,17 +4023,14 @@ private void _validateAutoIncrementValue(
 	}
 
 	private void _validateExternalReferenceCode(
-		String externalReferenceCode, long companyId, long groupId,
-		long objectEntryId, boolean newObjectEntry) {
+		String externalReferenceCode, long companyId, long objectDefinitionId,
+		long objectEntryId) {
 
-		ObjectEntry ercObjectEntries = objectEntryPersistence.fetchByERC_G_C(
-			externalReferenceCode, groupId, companyId);
+		ObjectEntry objectEntry = objectEntryPersistence.fetchByERC_C_ODI(
```

Comment:
@4lejandrito either this or not setting the groupId and companyId before makes some cases fail.
My first solution was to use both, fetchByERC_G_C and fetchByERC_C_ODI, however, leaving only fetchByERC_G_C worked for all the cases (but I understand that business rule of scoped by definitionId changes). I can keep investigate why the errors.

<img width="1618" alt="image" src="https://github.com/liferay-headless/liferay-portal/assets/35199951/60f13071-6558-42bb-9e46-3152fff97945">

Reply:
Hey @magjed4289 I just pushforced to move the call to `_setExternalReferenceCode` back down to make sure the objectDefinitionId is already populated (it was 0, that's why the right exception was not being thrown). With this change all the tests pass.